File: Deploy-Center-UI/src/types/DeployTask.ts
export type DeployTaskStatus =
  | "PENDING"
  | "UPLOADING"
  | "BUILDING"
  | "DEPLOYING"
  | "SUCCESS"
  | "FAILED"
  | "CANCELLED";

export type DeployTriggerType = "MANUAL" | "SCHEDULED";

export type DeployMechanism = "UPLOAD" | "CLOUD_BUILD";

export interface DeployTask {
  id: string;
  project_id: string;

  task_name?: string;

  status: DeployTaskStatus;

  trigger_type: DeployTriggerType;
  deploy_mechanism: DeployMechanism;

  upload_file_name?: string;
  upload_file_path?: string;

  build_image_name?: string;
  build_image_tag?: string;
  container_name?: string;

  dockerfile_content?: string;
  dockercommand_content?: string;

  started_at?: string;   // ISO datetime string
  finished_at?: string;  // ISO datetime string
  duration_ms?: number;

  failed_reason?: string;

  operator_name?: string;
  operator_id?: number;

  created_at?: string;   // ISO datetime string
  updated_at?: string;   // ISO datetime string
}

File: Deploy-Agent/src/models/entity/deploy_task.py
from pydantic import BaseModel
from typing import Optional, Literal
from datetime import datetime


class DeployTask(BaseModel):
    id: str
    project_id: str

    task_name: Optional[str] = None

    status: Literal[
        "PENDING",
        "RUNNING",
        "SUCCESS",
        "FAILED",
        "CANCELLED"
    ] = "PENDING"

    trigger_type: Literal["MANUAL", "SCHEDULED"] = "MANUAL"
    deploy_mechanism: Literal["UPLOAD", "CLOUD_BUILD"] = "UPLOAD"

    upload_file_name: Optional[str] = None
    upload_file_path: Optional[str] = None

    build_image_name: Optional[str] = None
    build_image_tag: Optional[str] = None
    container_name: Optional[str] = None
    
    dockerfile_content: Optional[str] = None
    dockercommand_content: Optional[str] = None

    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
    duration_ms: Optional[int] = None

    failed_reason: Optional[str] = None

    operator_name: Optional[str] = None  # 操作者名称（来自 Center 透传）
    operator_id: Optional[int] = None    # 操作者 ID（如有）

    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None